nemo/core/classes/modelPT.py

```diff
@@ -66,6 +77,55 @@ def __init__(self, cfg: DictConfig, trainer: Trainer = None):
             if 'test_ds' in self._cfg and self._cfg.test_ds is not None:
                 self.setup_test_data(self._cfg.test_ds)
 
+    def save_to(self, save_path: str):
+        """
+        Saves model instance (weights and configuration) into .nemo file. You can use "restore_from" method to fully
+        restore instance from .nemo file.
+
+        .nemo file is an archive (tar.gz) with the following:
+            model_config.yaml - model configuration in .yaml format. You can deserialize this into cfg argument for model's constructor
+            model_wights.chpt - model checkpoint
+
+        Args:
+            save_path: Path to .nemo file where model instance should be saved
+        """
+        with tempfile.TemporaryDirectory() as tmpdir:
+            config_yaml = path.join(tmpdir, _MODEL_CONFIG_YAML)
+            model_weights = path.join(tmpdir, _MODEL_WEIGHTS)
+            self.to_config_file(path2yaml_file=config_yaml)
+            trainer = self._trainer if self._trainer else Trainer()
+            if trainer.model != self:
+                trainer.model = self
+            trainer.save_checkpoint(filepath=model_weights, weights_only=True)
+            self.__make_nemo_file_from_folder(filename=save_path, source_dir=tmpdir)
+
+    @classmethod
+    def restore_from(cls, restore_path: str):
```

Comment:
Do we want to offer functionality to restore a checkpoint from just a simple pytorch checkpoint object , instead of a .nemo file? I understand .Nemo is just a container file, but this will significantly improve interop with pytorch environment.

Reply:
Downside of this is that unless you're really clever with inspection of tensor names, you need a tuple of [configuration/model shape, weights]. Or are you assuming the class is already full initialized appropriately (in which case, it wouldn't be a classmethod)?

Reply:
Good points @ryanleary. I think since these models are in fact just PT Modules under the hood, if a user is able to instantiate just the model using it's constructor (say by passing a config with just the components to instantiate the weights) , they can then simply point to a .pt checkpoint with self.load_state_dict().

Therefore to_config_dict and restore_from() become static methods to load everything from .Nemo files and user also has flexibility to use just the checkpoint as always. 

We should have test cases for this use case at the least. Otherwise this PR looks good to merge.

Reply:
Just discussed this with Oleksii... the restorefrom should also validate that the concrete class that is loading the checkpoint matches the `cls` field, and if not an exception can be thrown immediately (e.g. trying to restore_from on an ASR model with a BERT nemo file).

Reply:
This is a good static check. We will have to enforce the cls and params: pattern for all model components - say in this case encoder decoder and preprocessing, but thankfully we are all following this pattern at the moment anyway. 

Btw, won't the class constructor implicitly error out if such a mismatch occurs ? Primarily cause the keys in the constructor will not match the keys in the config.

Reply:
all of these are regular torch.nn.Module  so standard stuff like load_state_dict should work as expected.
added cls for model configs - but we'll probably want to polish it a bit more

Reply:
This check is what I originally did in NeMo 0.*. Both thumbs up.